src/System Application/App/File System/src/Scenario/FileScenarioImpl.Codeunit.al

```diff
@@ -0,0 +1,328 @@
+
+        if not FileScenarios.FindSet() then
+            exit;
+
+        // Find all valid scenarios. Invalid scenario may occur if the extension that added them was removed.
+        ValidFileScenarios := ValidFileScenarios.Hashtable();
+        foreach Scenario in Enum::"File Scenario".Ordinals() do
+            ValidFileScenarios.Add(Scenario, Scenario);
+
+        // Convert File Scenario-s to File Account Scenario-s so they can be sorted by "Display Name"
+        repeat
+            IsScenarioValid := ValidFileScenarios.Contains(FileScenarios.Scenario.AsInteger());
+
+            // Add entry for every scenario that exists and uses the file account. Skip the default scenario.
+            if (FileScenarios.Scenario <> Enum::"File Scenario"::Default) and IsScenarioValid then begin
+                FileAccountScenarios.Scenario := FileScenarios.Scenario.AsInteger();
+                FileAccountScenarios."Account Id" := FileScenarios."Account Id";
+                FileAccountScenarios.Connector := FileScenarios.Connector;
+                FileAccountScenarios."Display Name" := Format(FileScenarios.Scenario);
+
+                FileAccountScenarios.Insert();
+            end;
+        until FileScenarios.Next() = 0;
+
+        FileAccountScenarios.SetCurrentKey("Display Name"); // sort scenarios by "Display Name"
+    end;
+
+    local procedure AddEntry(var Result: Record "File Account Scenario"; EntryType: Option; Scenario: Integer; AccountId: Guid; Connector: Enum "File System Connector"; DisplayName: Text[2048]; Default: Boolean; var Position: Integer)
+    begin
+        // Add entry to the result while maintaining the position so that the tree represents the data correctly
+        Result.Init();
+        Result.EntryType := EntryType;
+        Result.Scenario := Scenario;
+        Result."Account Id" := AccountId;
+        Result.Connector := Connector;
+        Result."Display Name" := DisplayName;
+        Result.Default := Default;
+        Result.Position := Position;
+        Result.Insert();
+
+        Position := Position + 1;
+    end;
+
+    procedure AddScenarios(FileAccount: Record "File Account Scenario"): Boolean
+    var
+        FileScenario: Record "File Scenario";
+        SelectedScenarios: Record "File Account Scenario";
+        ScenariosForAccount: Page "File Scenarios For Account";
+    begin
+        FileAccountImpl.CheckPermissions();
+
+        if FileAccount.EntryType <> FileAccount.EntryType::Account then // wrong entry, the entry should be of type "Account"
+            exit;
+
+        ScenariosForAccount.Caption := StrSubstNo(ScenariosForAccountCaptionTxt, FileAccount."Display Name");
+        ScenariosForAccount.LookupMode(true);
+        ScenariosForAccount.SetRecord(FileAccount);
+
+        if ScenariosForAccount.RunModal() <> Action::LookupOK then
+            exit(false);
+
+        ScenariosForAccount.GetSelectedScenarios(SelectedScenarios);
```

Comment:
Similar to previous comment here in this file:
![image](https://github.com/user-attachments/assets/f45fb613-2ed7-4ffc-92bd-dcfd5523d4e2)

if ScenariosForAccount.GetSelectedScenarios is exit inside would trigger additional findset below, should be optimized to exit without additional SQL query

Reply:
Same here, Isntances are not the same. Temptable, no SQL involved.